src/Simplify.cpp

```diff
@@ -276,6 +276,24 @@ void Simplify::ScopedFact::learn_true(const Expr &fact) {
                 learn_lower_bound(v, i.bounds.min + 1);
             }
         }
+        const Min *min = lt->b.as<Min>();
+        if (min) {
+            // c < min(a, b) -> c < a, c < b
+            learn_true(lt->a < min->a);
+            learn_true(lt->a < min->b);
+            // c < min(a, b) -> !(a <= c), !(b <= c)
+            learn_false(min->a <= lt->a);
```

Comment:
Are the inverse statements required?

Reply:
Same question. Is this just to get a syntactic match on the <= form? The other cases here don't do it, but I think this is not exactly the same as the other cases.

Reply:
I found it was necessary (i.e. if I comment out the `learn_false()` lines, the test fails, since we can't eliminate the condition in the third specialization.  I added all the `learn_false()` clauses even though it may be true that only one or two are needed for this testcase.

We have a rewrite that changes `downsample = !(scale_factor_x > 1.0) && !(scale_factor_y > 1.0)` into `downsample = max(scale_factor_x, scale_factor_y) <= 1.0`.  Without these `learn_false()` clauses, the simplifier is unable to prove `max(scale_factor_x, scale_factor_y) <= 1.0 --> !(1.0 < scale_factor_x)` and similarly `max(scale_factor_x, scale_factor_y) <= 1.0 --> !(1.0 < scale_factor_y)`, so it can't trim the conditions in that branch.